DECLARING_DEPENDENCIES.md

```diff
@@ -0,0 +1,210 @@
+# Declaring Dependencies on Google Java Libraries
+
+Google maintains a number of open source Java libraries that make it
+easier to use services in Google Cloud Platform (GCP). Additionally Google
+maintains several foundational libraries that can be used for very
+general purposes, which the GCP libraries also depend on. A user of
+these libraries needs to use compatible versions of them in order to
+avoid dependency conflicts. This document explains how to use BOMs
+(bill of materials) to accomplish that and explains additional
+practices that make dependency management go more smoothly.
+
+These recommendations apply to the usage of the following libraries:
+
+- [guava](https://github.com/google/guava)
+- [protobuf](https://github.com/protocolbuffers/protobuf)
+- [grpc-java](https://github.com/grpc/grpc-java)
+- [gax-java](https://github.com/googleapis/gax-java)
+- [google-http-java-client](https://github.com/googleapis/google-http-java-client)
+- [google-oauth-java-client](https://github.com/googleapis/google-oauth-java-client)
+- [google-api-java-client](https://github.com/googleapis/google-api-java-client)
+- [google-cloud-java](https://github.com/googleapis/google-cloud-java)
+- [beam](https://github.com/apache/beam)
+
+## Ensuring Compatibility
+
+If you depend on one or more of these libraries, in order to
+ensure that your own project uses compatible versions of them, follow
+the guidance below for your build system.
+
+### Maven
+
+#### Use the requireUpperBoundDeps enforcer rule
+
+Dependency conflicts can happen with Maven even when libraries follow
```

Comment:
delete "Dependency conflicts can happen with Maven even when libraries follow
semver because"

Reply:
It seems a tad redundant, and  I'm  looking for any opportunities to cut the length. The shorter and more concise this is, the more people will read it.

Reply:
I definitely appreciate all of the recommendations to shorten the length. This instance seems to be in the tail of diminishing returns, however.

Reply:
Are we assuming here that the reader has read our other docs first? Or that they know what "semver" is?

Reply:
I'll add a link to the semver spec. As for our other docs, I'm not assuming the reader has read them. Do you have a specific recommendation?

Reply:
I still feel like semver is tangential to this discussion, and not something we can assume of the reader. This statement comes out of left field and is likely to kick the reader out because they're going to wonder:

A) What is semver?
B) Why does it matter here?

Then they go off and look for it to figure it out—it's worse with the link—and we lose them. 

Nor is semver an essential point here. We should simply drop the mention of it.

Reply:
Ok, removed.